nltk/draw/dispersion.py

```diff
@@ -15,51 +15,47 @@ def dispersion_plot(text, words, ignore_case=False, title="Lexical Dispersion Pl
     Generate a lexical dispersion plot.
 
     :param text: The source text
-    :type text: list(str) or enum(str)
+    :type text: list(str) or iter(str)
     :param words: The target words
     :type words: list of str
     :param ignore_case: flag to set if case should be ignored when searching text
     :type ignore_case: bool
     """
 
     try:
-        from matplotlib import pylab
+        import matplotlib.pyplot as plt
     except ImportError as e:
         raise ValueError(
             "The plot function requires matplotlib to be installed."
             "See https://matplotlib.org/"
         ) from e
 
-    text = list(text)
-    words.reverse()
```

Comment:
Reversing in place can lead to surprises when the same list of words is used in multiple dispersion plots (or other places besides a dispersion plot).